btf_types/src/lib.rs

```diff
@@ -95,18 +95,18 @@ impl fmt::Display for IntructionPosition {
     }
 }
 
-/// Provide structure to reprsent BF program.
+/// Provide structure to represent BF program.
 #[derive(Debug)]
 pub struct BrainFuckProgram {
     /// Name of the file from where program is parsed.
-    filename: String,
+    filename: PathBuf,
     /// List of instructions with location parsed from file.
     instructions: Vec<IntructionPosition>,
 }
 
 impl BrainFuckProgram {
     /// Create BF program based on the name of the file and it's content.
-    fn new(filename: String, content: String) -> Self {
+    fn new(filename: PathBuf, content: String) -> Self {
```

Comment:
A more idiomatic approach is to take `impl AsRef<Path>` rather than `PathBuf` directly.  You do this in `from_file` later in the code.

Reply:
maybe it is better to pass `&Path` to `new`, since in `from_file` I am doing this
```
let file_path_ref = file_path.as_ref();
let content = fs::read_to_string(file_path_ref)?;
```

Reply:
It's always better to use `&Path` rather than `PathBuf` and it's almost always better for this kind of `new()` type method to use `AsRef<Path>` rather than `&Path` for caller-ergonomics.

With `&Path` a caller who wants to use `"foo"` has to do `Path::new("foo")`